cou/apps/auxiliary.py

```diff
@@ -370,6 +375,41 @@ def _get_change_require_osd_release_step(self) -> PreUpgradeStep:
             coro=set_require_osd_release_option(ceph_mon_unit.name, self.model),
         )
 
+    def _get_wait_until_noout_set_step(self) -> PreUpgradeStep:
+        """Get wait until noout is set step.
+
+        :return: Step to wait until noout is set.
+        :rtype: PreUpgradeStep
+        """
+
+        async def _prompted_verification() -> None:
+            """Stop the upgrade and wait for noout is set.
+
+            :raises ApplicationError: When noout is unset
+            """
+            noout_set = False
+            progress_indicator.stop()
+            for i in range(1, self.retry_counter + 1):
```

Comment:
Can't we do this part with prompting the user if they agree to set the noout?

Reply:
What about running cou with `--auto-approve `? There should not be any prompt.

Reply:
It makes me wonder how `--auto-approve` works in unseal vault step as well

Reply:
Removed relevant code